core/src/server/rpc_module.rs

```diff
@@ -238,6 +256,37 @@ impl Debug for MethodKind {
 	}
 }
 
+/// A future that only resolves if a future was set.
+pub enum PendingOrFuture<'a> {
+	/// This will never resolve.
+	Pending,
+	/// Resolves once the future is completed.
+	Future(BoxFuture<'a, Result<(), DisconnectError>>),
+}
+
+impl<'a> std::fmt::Debug for PendingOrFuture<'a> {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		let msg = match self {
+			Self::Pending => "pending",
+			Self::Future(_) => "future",
```

Comment:
Very much a nit: I'd prolly stick close to the default Debug and spit out "Pending" and "Future(<future>)"

Reply:
the default doesn't work for `BoxFuture` stuff, that's the reason why :(

Reply:
Oh yeah, I just meant the actual strings being returned from the custom impl here could look like what would normally be generated :)

Reply:
ah gotcha